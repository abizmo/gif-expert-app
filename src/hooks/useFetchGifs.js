import { useEffect, useState } from "react";

import getGifs from '../helpers/getGifs'
const useFetchGifs = (category) => {
  const [gifs, setGifs] = useState([])
  const [loading, setLoading] = useState(true)

  useEffect(() => {
    getGifs( category )
      .then(setGifs)
      .catch(console.error)
      .finally(() => setLoading(false))
  }, [ category ])

  return [gifs, loading];
}

export default useFetchGifs;
