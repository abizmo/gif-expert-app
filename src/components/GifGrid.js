import React from 'react'
import PropTypes from 'prop-types'
import useFetchGifs from '../hooks/useFetchGifs'
import GifGridItem from './GifGridItem'

const GifGrid = ({ category }) => {
  const [ gifs, loading ] = useFetchGifs(category);

  return (
    <div>
      <h3>{ category }</h3>

      { loading && <p className="animate__animated animate__pulse">Loading...</p>}

      <div className="card-grid">
        {
          gifs.map((gif) => (
            <GifGridItem
              key={gif.id}
              { ...gif }
            />
          ))
        }
      </div>
    </div>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}

export default GifGrid
