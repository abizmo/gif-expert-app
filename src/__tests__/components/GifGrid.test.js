import { shallow } from 'enzyme'
import GifGrid from '../../components/GifGrid'
import useFetchGifs from '../../hooks/useFetchGifs'
jest.mock('../../hooks/useFetchGifs')

describe('GifGrid Tests', () => {
  const category = 'music'
  test('should render the component without items', () => {
    useFetchGifs.mockReturnValue([ [], true])
    const wrapper = shallow(<GifGrid category={category} />)
    expect(wrapper).toMatchSnapshot()
  })
  
  test('should render the component with items', () => {
    const gifs = [
      {
        id: '123',
        username: 'willy',
        title: 'puppy',
        img: 'http://localhost/puppy.jpg',
      }
    ]
    useFetchGifs.mockReturnValue([ gifs, false])
    const wrapper = shallow(<GifGrid category={category} />)

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('p').exists()).toBeFalsy()
    expect(wrapper.find('GifGridItem')).toHaveLength(gifs.length)
  })
  
})
