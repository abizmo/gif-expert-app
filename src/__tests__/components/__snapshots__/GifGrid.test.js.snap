// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid Tests should render the component with items 1`] = `
<div>
  <h3>
    music
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="123"
      img="http://localhost/puppy.jpg"
      key="123"
      title="puppy"
      username="willy"
    />
  </div>
</div>
`;

exports[`GifGrid Tests should render the component without items 1`] = `
<div>
  <h3>
    music
  </h3>
  <p
    className="animate__animated animate__pulse"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</div>
`;
